### PR TITLE
[Chore] Update xrefcheck to v0.2.2

### PR DESCRIPTION
**Pull request by alios**
## Description

Update xrefcheck to v0.2.2

## Related issue(s)

[xrefcheck issue 148](https://github.com/serokell/xrefcheck/issues/148)

#### Related changes (conditional)

- Tests (see [short guidelines](../blob/master/CONTRIBUTING.md#tests))
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - I checked whether I should update the docs and did so if necessary:
    - [ ] [README](../blob/master/README.md)
    - [ ] Haddock
    - [ ] [docs/](../blob/master/docs/)
  - [ ] I updated [changelog](../blob/master/ChangeLog.md) unless I am sure my changes are
        really minor.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../blob/master/docs/code-style.md).

### CODE REVIEW

File: .xrefcheck.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# Parameters of scanners for various file types.
+scanners:
+  markdown:
+    # Flavor of markdown, e.g. GitHub-flavor.
+    #
+    # This affects which anchors are generated for headers.
+    flavor: GitHub
```
Comment (dcastro) [inline .xrefcheck.yaml:11]: This file is not needed - `GitHub` is the default flavor

File: nix/sources.json [hunk 2]
```diff
@@ -55,16 +55,16 @@
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
```
Comment (dcastro) [inline nix/sources.json:55]: `xrefcheck` is mentioned twice in `pipeline.yaml`, let's have a look:

-----

  - label: crossref-verify
    if: *not_scheduled_autodoc
    commands:
    - nix run -f ci.nix xrefcheck -c
        xrefcheck --mode local-only --config ./.crossref-verifier.yaml
    soft_fail: true # TODO: remove
```

* The file `/.crossref-verifier.yaml` only contains default values, so we can delete it
* `--mode local-only` is not needed either`
* The CI step is named `crossref-verify`. This is the ooooold name of `xrefcheck` :P It has been renamed to "xrefcheck" a long time ago - so let's replace all references of the name "crossref-verify" with "xrefcheck"
* Let's remover `soft_fail: true`. The CI step is [in fact failing](https://buildkite.com/serokell/tezos-btc/builds/1081#01845296-d3bd-4a04-b3e3-5c80b5a0cd09), but we're not being told about it, so it goes unnoticed.

----

```
  - label: crossref-verify generated doc
    key: crossref-verify-generated
    if: *not_scheduled_autodoc
    depends_on:
      - contract-doc
      - contract-doc-master
    commands:
    - mkdir tmp
    - if [ "$BUILDKITE_BRANCH" = "master" ];
      then CONTRACT_DOC_STEP="contract-doc (master)";
      else CONTRACT_DOC_STEP="contract-doc";
      fi
    - buildkite-agent artifact download TZBTC-contract.md tmp/ --step "$$CONTRACT_DOC_STEP"
    - nix run -f ci.nix xrefcheck -c
        xrefcheck --mode local-only --config ./.crossref-verifier.yaml --root tmp
    soft_fail: true # TODO: remove
```

Basically, same items as above.

File: .xrefcheck.yaml (hunk 1 above)
Comment (alios) [reply .xrefcheck.yaml]: deleted it

File: nix/sources.json (hunk 2 above)
Comment (alios) [reply nix/sources.json:55]: implemented the mentioned fixes and fixed the broken links (in a second commit) as well.

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -124,7 +122,7 @@ steps:
     artifact_paths:
       - TZBTC-contract.md
 
-  - label: crossref-verify generated doc
+  - label: xrefcheck generated doc
     key: crossref-verify-generated
```
Comment (dcastro) [inline .buildkite/pipeline.yml:126]:
```suggestion
    key: xrefcheck-generated
```

**Review by dcastro — APPROVED** [review]

File: docs/code-style.md [hunk 4]
```diff
@@ -356,8 +356,8 @@ are partial functions.
 
 As usual, separate type classes with `, ` (comma and a space).
 
-If you are using GHC-8.2.2 or higher you *should* use
-[`-XDerivingStrategies`](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-strategies)
+If you are using GHC-8.2.1 or higher you *should* use
+[`-XDerivingStrategies`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/deriving_strategies.html)
```
Comment (lierdakil) [inline docs/code-style.md:360]: I don't think linking to the dev-version on gitlab instance is a good idea. Let's rather use a more specific link, e.g.

```suggestion
[`-XDerivingStrategies`](https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/deriving_strategies.html)
```

Or we could maybe even pin the version

```suggestion
[`-XDerivingStrategies`](https://downloads.haskell.org/ghc/9.4.1/docs/users_guide/exts/deriving_strategies.html)
```

File: docs/code-style.md [hunk 5]
```diff
@@ -386,7 +386,7 @@ You *should* avoid overly general signatures for functions that are actually
 used with only one type for each parameter. If you need the polymorhic version
 (i.e. if you are instantiating it more than once or if you are writing a
 library), you *may* use GHC's [`SPECIALIZE`
-pragma](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#specialize-pragma).
+pragma](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/pragmas.html#specialize-instance-pragma).
```
Comment (lierdakil) [inline docs/code-style.md:389]: Same here, either

```suggestion
pragma](https://downloads.haskell.org/ghc/9.4.1/docs/users_guide/exts/pragmas.html#specialize-instance-pragma).
```

or 

```suggestion
pragma](https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/pragmas.html#specialize-instance-pragma).
```

File: docs/code-style.md [hunk 6]
```diff
@@ -575,7 +575,7 @@ group independently.
 You *may* use [`weeder`][weeder] to detect
 unused dependencies and exported modules.
 
-[stylish-haskell]: https://github.com/serokell/metatemplates/blob/master/templates/haskell/.stylish-haskell.yaml
-[hlint]: https://github.com/serokell/metatemplates/blob/master/templates/haskell/.hlint.yaml
-[editorconfig]: https://github.com/serokell/metatemplates/blob/master/templates/haskell/.editorconfig
+[stylish-haskell]: https://raw.githubusercontent.com/serokell/metatemplates/master/haskell/.stylish-haskell.yaml
+[hlint]: https://raw.githubusercontent.com/serokell/metatemplates/master/haskell/.hlint.yaml
+[editorconfig]: https://raw.githubusercontent.com/serokell/metatemplates/master/.editorconfig
```
Comment (lierdakil) [inline docs/code-style.md:580]: Why are you linking raw files?

```suggestion
[stylish-haskell]: https://github.com/serokell/metatemplates/blob/master/haskell/.stylish-haskell.yaml
[hlint]: https://github.com/serokell/metatemplates/blob/master/haskell/.hlint.yaml
[editorconfig]: https://github.com/serokell/metatemplates/blob/master/.editorconfig
```

File: docs/code-style.md (hunk 4 above)
Comment (alios) [reply docs/code-style.md]: Gone for the `latest` version. Compiler fixed version imo only makes sense if we update those links everytime we switch the "preferred" compiler version...

File: docs/code-style.md (hunk 6 above)
Comment (alios) [reply docs/code-style.md]: done

File: docs/code-style.md (hunk 4 above)
Comment (lierdakil) [reply docs/code-style.md]: Docs for deriving strategies shouldn't really change (much) regardless of what compiler version we use ourselves, so there'd be no real reason to update these links. And there's a chance `latest` will eventually move the page somewhere and the link will break. But whatever, we'll cross that bridge when we get there, I guess.
